env0/resource_environment_test.go

```diff
@@ -1212,3 +1213,115 @@ func TestUnitEnvironmentResource(t *testing.T) {
 	testApiFailures()
 	testWaitFor()
 }
+
+func TestUnitTemplatelessEnvironmentResource(t *testing.T) {
+	resourceType := "env0_environment"
+	resourceName := "test"
+	accessor := resourceAccessor(resourceType, resourceName)
+
+	template := client.Template{
+		Id:          "id-template-0",
+		Name:        "template0",
+		Description: "description0",
+		Repository:  "env0/repo",
+		Path:        "path/zero",
+		Revision:    "branch-zero",
+		Retry: client.TemplateRetry{
+			OnDeploy: &client.TemplateRetryOn{
+				Times:      2,
+				ErrorRegex: "RetryMeForDeploy.*",
+			},
+			OnDestroy: &client.TemplateRetryOn{
+				Times:      1,
+				ErrorRegex: "RetryMeForDestroy.*",
+			},
+		},
+		Type:                 "terraform",
+		GithubInstallationId: 1,
+		TerraformVersion:     "0.12.24",
+		IsSingleUse:          true,
+	}
+
+	environment := client.Environment{
+		Id:                         "id0",
+		Name:                       "my-environment",
+		ProjectId:                  "project-id",
+		WorkspaceName:              "workspace-name",
+		TerragruntWorkingDirectory: "/terragrunt/directory/",
+		VcsCommandsAlias:           "alias",
+	}
+
+	createEnvironmentResourceConfig := func(environment client.Environment, template client.Template) string {
+		return fmt.Sprintf(`
+		resource "%s" "%s" {
+			name = "%s"
+			project_id = "%s"
+			workspace = "%s"
+			terragrunt_working_directory = "%s"
+			force_destroy = true
+			vcs_commands_alias = "%s"
+			template {
+				repository = "%s"
+				terraform_version = "%s"
+				type = "%s"
+				path = "%s"
+				revision = "%s"
+				retries_on_deploy = %d
+				retry_on_deploy_only_when_matches_regex = "%s"
+				retries_on_destroy = %d
+				retry_on_destroy_only_when_matches_regex = "%s"
+			}
+		}`,
+			resourceType, resourceName,
+			environment.Name,
+			environment.ProjectId,
+			environment.WorkspaceName,
+			environment.TerragruntWorkingDirectory,
+			environment.VcsCommandsAlias,
+			template.Repository,
+			template.TerraformVersion,
+			template.Type,
+			template.Path,
+			template.Revision,
+			template.Retry.OnDeploy.Times,
+			template.Retry.OnDeploy.ErrorRegex,
+			template.Retry.OnDestroy.Times,
+			template.Retry.OnDestroy.ErrorRegex,
+		)
+	}
+
+	t.Run("Success in create", func(t *testing.T) {
+		testCase := resource.TestCase{
+			Steps: []resource.TestStep{
+				{
+					Config: createEnvironmentResourceConfig(environment, template),
+					Check: resource.ComposeAggregateTestCheckFunc(
```

Comment:
This check does not run at the moment. This test is WIP and is primarily a sanity test that the plan is created without any errors.